src/models/necks/segmentation/unet_neck.py

```diff
@@ -0,0 +1,184 @@
+            out_channels: int,
+            use_attention: bool = False,
+            use_batchnorm: bool = True):
+        """Init DecoderBlock.
+
+        Args:
+            in_channels: Input channels.
+            skip_channels: Skip channels.
+            out_channels: Output channels.
+            use_attention: If ``True`` will use ``SEModule``.
+            use_batchnorm: If ``True``, ``BatchNormalisation`` layer between ``Conv2D`` and ``SEModule`` layers
+                is used.
+        """
+        super().__init__()
+
+        self.conv1 = ConvBnAct(
+            in_channels + skip_channels,
+            out_channels,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm)
+
+        self.attention1 = SCSEModule(channels=in_channels + skip_channels) if use_attention else None
+
+        self.conv2 = ConvBnAct(
+            out_channels,
+            out_channels,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm)
+
+        self.attention2 = SCSEModule(channels=out_channels) if use_attention else None
+
+    def forward(self, x: Tensor, skip: Optional[Tensor] = None) -> Tensor:
+        """Forward method.
+
+        Args:
+            x: Tensor for upsample.
+            skip: Skip-connection features.
+        """
+        x = F.interpolate(x, scale_factor=2, mode="nearest")
+
+        if skip is not None:
+            if skip.size(2) != x.size(2):
+                skip = F.interpolate(skip, size=(x.size(2), x.size(3)), mode="nearest")
+
+            x = torch.cat([x, skip], dim=1)
+
+            if self.attention1 is not None:
+                x = self.attention1(x)
+
+        x = self.conv1(x)
+        x = self.conv2(x)
+
+        if self.attention2 is not None:
+            x = self.attention2(x)
+
+        return x
+
+
+class CenterBlock(nn.Sequential):
+    """Optional building block for Unet like architecture."""
+    def __init__(self, in_channels: int, out_channels: int, use_batchnorm: int = True):
+        """Init CenterBlock.
+
+        Args:
+            in_channels: Input channels.
+            out_channels: Output channels.
+            use_batchnorm: If True then `ConvBnAct` will use batch normalization.
+        """
+        conv1 = ConvBnAct(
+            in_channels,
+            out_channels,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm)
+        conv2 = ConvBnAct(
+            out_channels,
+            out_channels,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm)
+        super().__init__(conv1, conv2)
+
+
+@NECKS.register_class
+class UnetNeck(BaseModel):
+    """Unet is a fully convolution neural network for image semantic segmentation.
+    Paper: https://arxiv.org/pdf/1505.04597.
+    """
+
+    def __init__(self,
+                 encoder_channels: Tuple[int],
+                 decoder_channels: Tuple[int] = (512, 256, 128, 64, 32),
```

Comment:
Why (512, 256, 128, 64, 32) but not (512, 256, 128, 64, 64)  like Resnet return?